src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -5903,6 +5904,60 @@ mesh::field::verify(const Node &field,
     return res;
 }
 
+//-----------------------------------------------------------------------------
+void
+mesh::field::generate_strip(Node& fields,
+                            const std::string& toponame,
+                            const std::string& dest_toponame)
+{
+    Node newfields;
+
+    NodeConstIterator fields_it = fields.children();
+    while (fields_it.has_next())
+    {
+        const Node& field = fields_it.next();
+
+        if (field["topology"].as_string() == toponame)
+        {
+            // TODO Something useful with grid functions, when needed by users.
+            if (field.has_child("association"))
+            {
+                if (field["association"].as_string() == "element")
+                {
+                    const std::string newfieldname = dest_toponame + "_" + fields_it.name();
+                    Node& newfield = newfields[newfieldname];
+
+                    newfield["association"] = "element";
+                    newfield["topology"] = dest_toponame;
+                    if (field.has_child("volume_dependent"))
+                        newfield["volume_dependent"] = field["volume_dependent"].as_string();
+                    if (field.has_child("matset"))
+                    {
+                        newfield["matset"] = field["matset"].as_string();
+                        newfield["matset_values"].set_external(field["matset_values"]);
+                    }
+                    if (field.has_child("values"))
+                    {
+                        newfield["values"].set_external(field["values"]);
+                    }
+                }
+                else
+                {
+                    // For now, do nothing.
+                    // TODO Something useful with vertex fields.  Confer with users.
+                }
+            }
+        }
+    }
+
+    NodeConstIterator newfields_it = newfields.children();
+    while (newfields_it.has_next())
+    {
+        const Node& newfield = newfields_it.next();
+        fields[newfields_it.name()] = newfield;
+    }
```

Comment:
I used the pattern
- iterate over source Node's children, generating a new from qualifying Nodes
- add the generated Nodes back to the target, which happens to be the source
because the source and target are the same nodes (`mesh["fields"]`).

If there's a better pattern (like hypothetical `Node::MergeFrom(const Node& other)`), then we can use that.